src/bp/ui-studio/src/web/reducers/flows.ts

```diff
@@ -176,24 +187,31 @@ const copyName = (siblingNames, nameToCopy) => {
   }
 }
 
-const doRenameFlow = ({ flow, name, flows }) =>
+const doRenameFlow = ({ currentName, newName, flows }) =>
   flows.reduce((obj, f) => {
-    if (f.name === flow) {
-      f.name = name
-      f.location = name
+    if (f.name === currentName) {
+      f.name = newName
+      f.location = newName
     }
 
     if (f.nodes) {
       let json = JSON.stringify(f.nodes)
-      json = json.replace(flow, name)
+      json = json.replace(currentName, newName)
       f.nodes = JSON.parse(json)
     }
 
-    obj[f.name] = f
+    const newObj = { ...obj }
+    newObj[f.name] = f
 
-    return obj
+    return newObj
   }, {})
 
+const doDeleteFlow = ({ name, flowsByName }) => {
+  flowsByName = _.omit(flowsByName, name)
+  const flows = _.values(flowsByName)
+  return doRenameFlow({ currentName: name, newName: '', flows })
+}
```

Comment:
Why is delete renaming ?

Reply:
It renames all of the deleted flow occurences to an empty string.